checkbox-ng/checkbox_ng/launcher/test_agent.py

```diff
@@ -61,3 +66,132 @@ def test_invoked_ok(
         # the server was started
         self.assertTrue(server.start.called)
+
+    # used to load an empty launcher with no error
+    def test_invoked_with_resume_warns(
+        self,
+        getenv_mock,
+        geteuid_mock,
+        threaded_server_mock,
+        remote_assistant_mock,
+        remote_strategy_mock,
+        session_assistant_mock,
+        pwdless_sudo_mock,
+        logger_mock,
+        gettext_mock,
+    ):
+        ctx_mock = mock.MagicMock()
+        ctx_mock.args.resume = True
+
+        self_mock = mock.MagicMock()
+
+        geteuid_mock.return_value = 0  # agent will not run as non-root
+        pwdless_sudo_mock.return_value = True  # agent needs pwd-less sudo
+        getenv_mock.return_value = None
+
+        with mock.patch("builtins.open"):
+            RemoteAgent.invoked(self_mock, ctx_mock)
+
+        # agent server was created
+        self.assertTrue(threaded_server_mock.called)
+        server = threaded_server_mock.return_value
+        # the server was started
+        self.assertTrue(server.start.called)
+        self.assertTrue(logger_mock.warning.called)
+
+    def test_invoked_with_session(
+        self,
+        getenv_mock,
+        geteuid_mock,
+        threaded_server_mock,
+        remote_assistant_mock,
+        remote_strategy_mock,
+        sa_mock,
+        pwdless_sudo_mock,
+        logger_mock,
+        gettext_mock,
+    ):
+        ctx_mock = mock.MagicMock()
+        ctx_mock.args.resume = False
+
+        self_mock = mock.MagicMock()
+
+        geteuid_mock.return_value = 0
+
+        ctx_mock.sa.get_resumable_sessions.return_value = [
+            ResumeCandidate("an_id", SessionMetaData())
+        ]
+        with mock.patch(
+            "checkbox_ng.launcher.agent.is_the_session_noninteractive"
+        ) as itni:
+            itni.return_value = True
+
+            RemoteAgent.invoked(self_mock, ctx_mock)
+            self.assertTrue(
+                remote_assistant_mock.return_value.resume_by_id.called
+            )
+
+        remote_assistant_mock.reset_mock()
+
+        with mock.patch(
+            "checkbox_ng.launcher.agent.is_the_session_noninteractive"
+        ) as itni:
+            itni.return_value = False
+
+            RemoteAgent.invoked(self_mock, ctx_mock)
+            self.assertFalse(
+                remote_assistant_mock.return_value.resume_by_id.called
+            )
+
+
+class IsTheSessionNonInteractiveTests(TestCase):
+    def test_a_non_interactive_one(self):
+        a_non_interactive_launcher = """
+            [ui]
+            type = silent
+        """
+        app_blob = json.dumps({"launcher": a_non_interactive_launcher})
+        metadata = SessionMetaData(app_blob=app_blob.encode("utf-8"))
+
+        candidate = ResumeCandidate("an_id", metadata)
+        self.assertTrue(is_the_session_noninteractive(candidate))
+
+    def test_an_interactive(self):
+        a_non_interactive_launcher = ""  # the defautl one is interactive
```

Comment:
This also tests the default in a way, I don't think this is a good idea, just make it interactive

Reply:
Very valid point. Fixed.